### PR TITLE
Add append strategy for Dataframe operator

### PR DESCRIPTION
**Pull request by sunank200**
# Description
## What is the current behavior?

Insert into an existing table (output table) from a Dataframe? It seems that if_exists is always set to replace: https://github.com/astronomer/astro-sdk/blob/1.3.0/python-sdk/src/astro/sql/operators/dataframe.py#L188

Refer to https://astronomer.slack.com/archives/C02B8SPT93K/p1670370753724899

closes: #1379

## What is the new behaviour?

- Introduce a parameter `if_exists" 
- Add tests for append

## Does this introduce a breaking change?
No

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: python-sdk/src/astro/sql/operators/dataframe.py [hunk 1]
```diff
@@ -119,6 +120,7 @@ def __init__(
         database: str | None = None,
         schema: str | None = None,
         columns_names_capitalization: ColumnCapitalization = "original",
+        if_exists: LoadExistStrategy | None = "replace",
```
Comment (kaxil) [inline python-sdk/src/astro/sql/operators/dataframe.py:123]:
```suggestion
        if_exists: LoadExistStrategy = "replace",
```

File: python-sdk/src/astro/sql/operators/dataframe.py [hunk 2]
```diff
@@ -133,6 +135,7 @@ def __init__(
             self.output_table = None
         self.op_args = self.kwargs.get("op_args", ())  # type: ignore
         self.columns_names_capitalization = columns_names_capitalization
+        self.if_exists = if_exists if if_exists else "replace"
```
Comment (kaxil) [inline python-sdk/src/astro/sql/operators/dataframe.py:138]:
```suggestion
        self.if_exists = if_exists
```

Comment (kaxil) [reply python-sdk/src/astro/sql/operators/dataframe.py:138]: Will it ever return `None`?

File: python-sdk/src/astro/sql/operators/dataframe.py (hunk 1 above)
Comment (sunank200) [reply python-sdk/src/astro/sql/operators/dataframe.py:123]: Wouldn't it break if the user doesn't specify `if_exists` for users in the older version of Astro-SDK?

File: python-sdk/src/astro/sql/operators/dataframe.py (hunk 2 above)
Comment (sunank200) [reply python-sdk/src/astro/sql/operators/dataframe.py:138]: Wouldn't it break if the user doesn't specify `if_exists` for users in the older version of Astro-SDK? that's the reason I kept it optional

File: python-sdk/src/astro/sql/operators/dataframe.py (hunk 1 above)
Comment (kaxil) [reply python-sdk/src/astro/sql/operators/dataframe.py:123]: `"replace"` is already there as the default

File: python-sdk/src/astro/sql/operators/dataframe.py (hunk 2 above)
Comment (kaxil) [reply python-sdk/src/astro/sql/operators/dataframe.py:138]: Can you give an example of that please?

File: python-sdk/src/astro/sql/operators/dataframe.py (hunk 1 above)
Comment (sunank200) [reply python-sdk/src/astro/sql/operators/dataframe.py:123]: Yes, I tired it locally now. While using annotations, even when the value is not passed at all it picks up the default value.

**Review by kaxil — APPROVED** [review]

**Pull request merged**